File: api/src/sqs.js
const { SQSClient, SendMessageCommand } = require("@aws-sdk/client-sqs");

const client   = new SQSClient({ region: process.env.AWS_REGION || "us-east-1" });
const QUEUE_URL = process.env.SQS_QUEUE_URL;

/**
 * Publish a structured event to the SQS queue.
 * Silently skips if SQS_QUEUE_URL is not configured (e.g. local dev).
 */
async function publishEvent(eventType, payload) {
  if (!QUEUE_URL) return;

  const message = {
    eventType,
    payload,
    timestamp: new Date().toISOString(),
    source:    "lks-api",
  };

  await client.send(new SendMessageCommand({
    QueueUrl:    QUEUE_URL,
    MessageBody: JSON.stringify(message),
    MessageAttributes: {
      EventType: { DataType: "String", StringValue: eventType },
    },
  }));

  console.log(`SQS event published: ${eventType}`);
}

module.exports = { publishEvent };

File: api/sqs.py
import os
import json
import logging
import time
import boto3
from botocore.exceptions import ClientError

log = logging.getLogger(__name__)
_client = None
QUEUE_URL = os.getenv("SQS_QUEUE_URL", "")


def _get_client():
    global _client
    if _client is None:
        _client = boto3.client("sqs", region_name=os.getenv("AWS_REGION", "us-east-1"))
    return _client


def publish_event(event_type: str, payload: dict):
    """Publish event to SQS. Silently skips if QUEUE_URL not configured."""
    if not QUEUE_URL:
        return
    try:
        message = json.dumps({
            "eventType": event_type,
            "payload": payload,
            "timestamp": time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime()),
            "source": "lks-api",
        })
        _get_client().send_message(
            QueueUrl=QUEUE_URL,
            MessageBody=message,
            MessageAttributes={
                "EventType": {"DataType": "String", "StringValue": event_type},
            },
        )
        log.info(f"SQS event published: {event_type}")
    except ClientError as e:
        log.error(f"SQS publish failed: {e}")
